usr.bin/du/du.c

```diff
@@ -453,9 +466,10 @@ linkchk(FTSENT *p)
 		/* Pull a node from the free list if we can. */
 		le = free_list;
 		free_list = le->next;
-	} else
+	} else {
 		/* Malloc one if we have to. */
 		le = malloc(sizeof(struct links_entry));
+	}
 	if (le == NULL) {
 		stop_allocating = 1;
 		warnx("No more memory for tracking hard links");
```

Comment:
I'm using df as a template.  It didn't look like they converted the warnx on malloc failures.  Does xo_warnx allocate?  I figured it might and that is why df does it the way it does, but maybe I should check for sure.

Reply:
By default I believe that xo_warnx() etc. just output unstructured data to standard error, and they don't appear to allocate memory. (And libxo will print an error if it hits an internal memory allocation failure.)

Reply:
In tree examples seem somewhat inconsistent here so I don't have any issues converting the rest of the calls.